From: Randy Dunlap <rdunlap@infradead.org>
Date: Thu, 2 Apr 2015 17:10:55 -0700
Subject: usb/misc: fix chaoskey build, needs HW_RANDOM
Origin: https://git.kernel.org/linus/d9aab404e60d122ded979fa0b81db42fb680d867
Bug-Debian: https://bugs.debian.org/839616

Fix build errors when HW_RANDOM is not enabled:

drivers/built-in.o: In function `chaoskey_disconnect':
chaoskey.c:(.text+0x5f3f00): undefined reference to `hwrng_unregister'
drivers/built-in.o: In function `chaoskey_probe':
chaoskey.c:(.text+0x5f42a6): undefined reference to `hwrng_register'

Signed-off-by: Randy Dunlap <rdunlap@infradead.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/usb/misc/Kconfig | 1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/misc/Kconfig
+++ b/drivers/usb/misc/Kconfig
@@ -251,6 +251,7 @@ config USB_HSIC_USB3503
 
 config USB_CHAOSKEY
 	tristate "ChaosKey random number generator driver support"
+	depends on HW_RANDOM
 	help
 	  Say Y here if you want to connect an AltusMetrum ChaosKey to
 	  your computer's USB port. The ChaosKey is a hardware random
